File: Easy/21. Merge Two Sorted Lists/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
        result = temp = ListNode(0)
        
        while l1 != None and l2 != None:
            if l1.val >= l2.val:
                temp.next = l2
                l2 = l2.next
                
            elif l1.val < l2.val:
                temp.next = l1
                l1 = l1.next
                
            temp = temp.next
                
        if l1 == None:
            temp.next = l2
        elif l2 == None:
            temp.next = l1
            
        return result.next
                

File: Easy/21. Merge Two Sorted Lists/solution.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* mergeTwoLists(struct ListNode* l1, struct ListNode* l2){
    struct ListNode* temp = calloc(1, sizeof(struct ListNode));
    struct ListNode* result = calloc(1, sizeof(struct ListNode));
    result = temp;
    
    while(l1 != NULL && l2 != NULL){
        if(l1->val > l2->val){
            temp->next = l2;
            l2 = l2->next;
        }
        else if(l1->val <= l2->val){
            temp->next = l1;
            l1 = l1->next;
        }
        temp = temp->next;
    }
    
    if(l1 == NULL) temp->next = l2;
    else if(l2 == NULL) temp->next = l1;
    
    return result->next;
}